src/program/downloaders/alldebrid.py

```diff
@@ -51,594 +34,185 @@ def validate(self) -> bool:
-            logger.error("All-Debrid movie filesize max is not set or invalid.")
-            return False
-        if not isinstance(self.download_settings.episode_filesize_min, int) or self.download_settings.episode_filesize_min < -1:
-            logger.error("All-Debrid episode filesize min is not set or invalid.")
-            return False
-        if not isinstance(self.download_settings.episode_filesize_max, int) or self.download_settings.episode_filesize_max < -1:
-            logger.error("All-Debrid episode filesize max is not set or invalid.")
-            return False
         if self.settings.proxy_enabled and not self.settings.proxy_url:
             logger.error("Proxy is enabled but no proxy URL is provided.")
             return False
         try:
-            response = ping(
-                f"{AD_BASE_URL}/user?{AD_PARAM_AGENT}",
-                additional_headers=self.auth_headers,
-                proxies=self.proxy,
-                specific_rate_limiter=self.inner_rate_limit,
-                overall_rate_limiter=self.overall_rate_limiter)
-            if response.is_ok:
-                user_info = response.data.data.user
-                expiration = user_info.premiumUntil or 0
+            user_info = get_user()
+            if user_info:
+                user = user_info.get("data", {}).get("user", {})
+                expiration = user.get("premiumUntil", 0)
                 expiration_datetime = datetime.utcfromtimestamp(expiration)
-                time_left = expiration_datetime - datetime.utcnow()
-                days_left = time_left.days
-                hours_left, minutes_left = divmod(time_left.seconds // 3600, 60)
-                expiration_message = ""
-    
-                if days_left > 0:
-                    expiration_message = f"Your account expires in {days_left} days."
-                elif hours_left > 0:
-                    expiration_message = f"Your account expires in {hours_left} hours and {minutes_left} minutes."
-                else:
-                    expiration_message = "Your account expires soon."
-    
-                if not user_info.isPremium or False:
+                expiration_message = premium_days_left(expiration_datetime)
+
+                premium = bool(user.get("isPremium", False))
+                if not premium:
                     logger.error("You are not a premium member.")
                     return False
                 else:
                     logger.log("DEBRID", expiration_message)
-    
-                return user_info.isPremium or False
+
+                return premium
         except ConnectTimeout:
             logger.error("Connection to All-Debrid timed out.")
         except Exception as e:
             logger.exception(f"Failed to validate All-Debrid settings: {e}")
         return False
 
-    def run(self, item: MediaItem) -> bool:
```

Comment:
the run func should stay as its used everywhere

Reply:
It looks like this was refactored in #665 from [here](https://github.com/rivenmedia/riven/pull/665/files#diff-9fe4cd87fe5a74efd7de63e59a9bb11128bded724aca055c3f5d007cc266ff93L108) to [here](https://github.com/rivenmedia/riven/pull/665/files#diff-6a86fa88ec0fa85d273a3521a97cec84d1032da12b553d02cce720493abf62a1L32). I've made sure all the functions from `RealDebridDownloader` are also implemented here - they should conform to the same interface

Reply:
ah yes, I forgot we ditched the run() in the RD refactor. This looks great!